tests/Specs/CIFSClient.cs

```diff
@@ -82,5 +81,82 @@ public void SharedFolder_GetFile_ShouldReturnFileStream_WhenFileExists()
             Assert.NotNull(result);
             Assert.Equal(new Uri(@"\\testServer\testShare\Folder\testFile.txt"), result.Item1);
         }
+
+        // TODO Add test method for UNhappy path of PutFile
+
+        [Fact]
+        public void SharedFolder_PutFile_ShouldWriteFileToSharedFolder()
+        {
+            // Arrange
+            var mockFileStore = new Mock<ISMBFileStore>();
+            var localFilePath = "localTestFile.txt";
+            var localFileData = "Test data";
+            var remoteFilePath = "remoteTestFile.txt";
+
+            // Create a temporary local file for testing
+            File.WriteAllText(localFilePath, localFileData);
```

Comment:
If we use the FileSystem abstractions on the CIFSClient, we could avoid having to write to the physical file system here, which is always a plus 😁